### PR TITLE
openafs: fix OpenAFS 1.6 on Linux 4.20

### PR DESCRIPTION
**Pull request by artemist**
###### Motivation for this change
I have been requested to add linuxPackages_latest support to OpenAFS 1.6. I found working patches, although not enough for 5.0

###### Things done
This patch cherry-picks commits from the OpenAFS git repository using
`git show` to add support for Linux 4.20 for OpenAFS 1.6.

There do not seem to be patches in the Git repository for full Linux 5.0
support, although I have added the ones which exist so far.

This patch has been tested on Linux 4.4, 4.14, 4.20, and is confirmed to
fail on Linux 5.0. It does not change the drv of the openafs package,
only linuxPackages_*.openafs.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by veprbl** [comment]
cc @spacefrogg

**Comment by artemist** [comment]
@GrahamcOfBorg build linuxPackages_4_20.openafs

**Comment by veprbl** [comment]
@GrahamcOfBorg build linuxPackages_4_20.openafs

**Comment by veprbl** [comment]
@GrahamcOfBorg build linuxPackages_4_20.openafs

File: pkgs/servers/openafs/1.6/default.nix [hunk 1]
```diff
@@ -14,7 +14,7 @@ stdenv.mkDerivation rec {
 
   buildInputs = [ ncurses ];
 
-  patches = stdenv.lib.optional (tsmbac != null) ./tsmbac.patch;
+  patches = stdenv.lib.optional (tsmbac != null) ./patches/tsmbac.patch;
```
Comment (veprbl) [inline pkgs/servers/openafs/1.6/default.nix]: Perhaps lets not move this for now?

**Review by veprbl — APPROVED** [review]
Builds against most kernels (except hardened). Works.

**Comment by spacefrogg** [comment]
@artemist Thanks for the effort! Looks good from my side.

**Comment by artemist** [comment]
@veprbl That's an annoying issue with struct layout randomization. i'm writing a patch I'll submit upstream after some testing, but it's out of scope for this PR (and quite possibly nixpkgs)

**Pull request merged**